sdks/python/apache_beam/io/aws/s3io.py

```diff
@@ -118,7 +118,18 @@ def list_prefix(self, path):
     logging.info("Starting the size estimation of the input")
 
     while True:
-      response = self.client.list(request)
+
+      #The list operation will raise an exception when trying to list a nonexistent S3 path. 
+      #This should not be an issue here. 
+      #Ignore this exception or it will break the procedure.
+      try:
+        response = self.client.list(request)
+      except messages.S3ClientError as e:
+        if e.code == 404:
+          break
```

Comment:
Should the response be some kind of empty response? If we just break out here, `response` will be `None`, and `response.items` will raise an exception, right?

Reply:
@icemoon1987 fyi

Reply:
Hi pabloem,

Thank you for reviewing :)

This "break" will exit the "while" loop start at line 120. The "response.items" operation in line 133 is also in the "while" loop in line 120. So the "response.items" (line 133) will not raise an exception. The variable "response" is only used in the "while" loop start at line 120.

I directly "break" here because I saw the "break" operation in line 142. They should have the similar effect.

The "break" here will stop the loop and the function: list_prefix will return the variable "file_sizes" in line 147. Since the variable "file_sizes" is initialized as an empty dict in line 114. The list_prefix function will return an empty dict. It should be safe.

Reply:
You are correct. Thanks for pointing that out. FYI, it seems that there are a few linnt errors and a few errors in precommit tests.